app/index.js

```diff
@@ -9,98 +9,60 @@ class BlockstackGenerator extends Generator {
-      default: true
-    }];
+      // Have Yeoman greet the user.
+      this.log(yosay(
+        'Welcome to the ' + chalk.red('Blockstack') + ' app generator!'
+      ));
 
-    return this.prompt(prompts).then(function (props) {
-      // To access props later use this.props.someAnswer
-      this.props = props;
-    }.bind(this));
+      const answers = await this.prompt([{
+        type: 'list',
+        name: 'generator_type',
+        message: 'Select Blockstack app output type',
+        choices: [
+          { name: 'React', value: 'react' },
+          { name: 'Plain JavaScript', value: 'plain' },
+          { name: 'Vue', value: 'vue' }
+        ]
+      }]);
+      this.outputAppType = answers['generator_type'];
+    }
   }
 
-  writing() {
-    this.fs.copy(
-      this.templatePath('_package.json'),
-      this.destinationPath('package.json')
-    );
-    this.fs.copy(
-      this.templatePath('editorconfig'),
-      this.destinationPath('.editorconfig')
-    );
-    this.fs.copy(
-      this.templatePath('gitignore'),
-      this.destinationPath('.gitignore')
-    );
-    this.fs.copy(
-      this.templatePath('requires.js'),
-      this.destinationPath('requires.js')
-    );
-    this.fs.copy(
-      this.templatePath('firebase.json'),
-      this.destinationPath('firebase.json')
-    );
-    // server
-    this.fs.copy(
-      this.templatePath('server.js'),
-      this.destinationPath('server.js')
-    );
-    // styles
-    this.fs.copy(
-      this.templatePath('public/app.css'),
-      this.destinationPath('public/app.css')
-    );
-    this.fs.copy(
-      this.templatePath('public/bootstrap.min.css'),
-      this.destinationPath('public/bootstrap.min.css')
-    );
-    // scripts
-    this.fs.copy(
-      this.templatePath('public/app.js'),
-      this.destinationPath('public/app.js')
-    );
-    // images
-
-    this.fs.copy(
-      this.templatePath('public/favicon.ico'),
-      this.destinationPath('public/favicon.ico')
-    )
-
-    this.fs.copy(
-      this.templatePath('public/index.html'),
-      this.destinationPath('public/index.html')
-    );
-    // publicExtras
-    this.fs.copy(
-      this.templatePath('public/robots.txt'),
-      this.destinationPath('public/robots.txt')
-    );
-    this.fs.copy(
-      this.templatePath('public/manifest.json'),
-      this.destinationPath('public/manifest.json')
-    );
-    this.fs.copy(
-      this.templatePath('public/white-logo.svg'),
-      this.destinationPath('public/white-logo.svg')
-    );
+  main() {
+    if (this.outputAppType === 'react') {
+      return this.composeWith(require.resolve('../react'));
+    } else if (this.outputAppType === 'plain') {
+      return this.composeWith(require.resolve('../webpack'));
+    } else if (this.outputAppType === 'vue') {
+      return this.composeWith(require.resolve('../vue'));
+    } else {
+      throw new Error(`Unexpected app output type selected: ${this.outputAppType}`);
+    }
```

Comment:
Conflict message is throwing a string of characters when a `package.json` conflict exists. (Nice to throw an error but unclear the choices)

![image](https://user-images.githubusercontent.com/1347209/64715949-46c42580-d476-11e9-9620-e15057ef3405.png)

LOoks like user needs to press a key to see what their choices are wrt these.  Would be helpful to say:

```
 ? Overwrite package.json? (Choose one of y/n/a/x/d or H to display all the options)
```

Reply:
This should not be new behavior introduce by this PR. It's what yo does by default when package gen is performed within a directory that already has the files. Is something different happening when running with the current version? 


Also, I personally don't like yo defaulting to this behavior, and would be open to changing it. In the clarity-sdk (which is setup using integrated yo in a similar way), I added logic for `if currently directory is not empty, then create a new directory and output app into that`. 
That's also why with the clarity sdk you can specify the output directly in the command, e.g. `npx create-clarity-dev {my_app_directory}`. 

Do you think we should we add the same capabilities?

Reply:
The reason for the overwrite was not confusing for me.  My suggestion was in regard to the `(ynaxdH)` help prompt. Your solution for Clarity would work here too, do you think it would be quick to add?

Reply:
It wouldn't be quick enough (or a small enough change) to fit into this PR, but I think we should open an issue for it. I think we'd also need to open an issue for trying to customize the overwrite help messaging.